### PR TITLE
USAGOV-1056: Store image digest info in annotated git tags

### PR DESCRIPTION
**Pull request by ghost**
## Jira Task

https://cm-jira.usa.gov/browse/USAGOV-1056

## Description

In order to reliably/programmatically obtain the latest production images for use in DR situations, add scripts to:

1. Add digest information to an annotated git tag
2. Find the latest annotated tag and extract digest information from it

My Initial thoughts on how to do this:

1. Capture image digest information at SUCCESSFUL deployment time (manual AND CircleCI?)
2. Tag the head of the production branch at SUCCESSFUL deployment time, and apply an annotated tag containing digest information.

If the format of the tag name and annotation are standardized, we can (in a DR scenario) find the latest correctly formatted tag and query the annotation for digest information, and use the digests to re-deploy the correct images to production.

## Checklist for the Developer

- [x] A link to the JIRA ticket has been added above.
- [x] The JIRA ticket identifies the desired result of this change.
- [ ] The JIRA ticket contains clear acceptance criteria.
- [ ] The JIRA ticket contains clear testing steps.
- [ ] The JIRA ticket contains a description of "Done".
- [ ] Any preparation/installation/update steps required for this code to work properly (drush commands, scripts, configuration updates) are documented in the ticket.

## Checklist for the Peer Reviewers
- [ ] The branch name of this PR matches the project standards.
- [ ] QA steps are followed and any changes to process are updated in the ticket.
- [ ] Code Standards are followed, there are no bad practices in use.
- [ ] Config changes (if any) include only necessary changes.
- [ ] No errors in Drupal or Client Browser.
- [ ] Code has been tested locally (if possible).
- [ ] Following AC and Testing Steps verify changes work as expected.
- [ ] There are no known side-effects outside of expected behavior.

### CODE REVIEW

**Comment by ghost** [comment]
The main changes to review are:
1. `.circleci/config/.yml`  (note that the approval filters for branch - USAGOV-1056-store-image-digests will be deleted before merge to dev.
2. `bin/deploy/includes` Added several functions to assert the existence of various Cloud Foundry entities.

The new files can't be tested fully until this moves to stage.  So if the above 2 files are syntactically correct, we should be good to merge to dev (after I remove the branch-specific approval filters).

**NOTE:** _If approved, please do not merge yet.  I need to remove the branch approval filters first._

**Review by jacobaaronyeager — DISMISSED** [review]

**Pull request merged**
